### PR TITLE
ref(js): Remove organization.projects usage from getOnboardingTasks

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Review by billyvg — APPROVED** [review]

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 1]
```diff
@@ -17,25 +17,33 @@ import {
...
 
-function hasPlatformWithSourceMaps(organization: Organization): boolean {
-  const projects = organization?.projects;
-  if (!projects) {
-    return false;
-  }
-
-  return projects.some(({platform}) => platform && sourceMaps.includes(platform));
+function hasPlatformWithSourceMaps(projects: Project[] | undefined) {
+  return projects !== undefined
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/taskConfig.tsx:22]: should we only check if `project` or `!!project `?

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 2]
```diff
@@ -60,22 +68,21 @@ export function getOnboardingTasks(
...
-              eventType="error"
-              onIssueReceived={() => !taskIsDone(task) && onCompleteTask()}
-            >
-              {() => <EventWaitingIndicator />}
-            </EventWaiter>
-          ) : null
-        )
+      SupplementComponent: withApi(({api, task, onCompleteTask}: FirstEventWaiterProps) =>
+        projects !== undefined &&
+        projects.length > 0 &&
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/taskConfig.tsx:73]:
```suggestion
        !!projects?.length &&
```

File: static/app/components/onboardingWizard/taskConfig.tsx [hunk 3]
```diff
@@ -113,22 +120,21 @@ export function getOnboardingTasks(
...
-              eventType="transaction"
-              onIssueReceived={() => !taskIsDone(task) && onCompleteTask()}
-            >
-              {() => <EventWaitingIndicator />}
-            </EventWaiter>
-          ) : null
-        )
+      SupplementComponent: withApi(({api, task, onCompleteTask}: FirstEventWaiterProps) =>
+        projects !== undefined &&
+        projects.length > 0 &&
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/taskConfig.tsx:125]:
```suggestion
            !!projects?.length &&
```

**Review by priscilawebdev — APPROVED** [review]
left some suggestions but it looks good to me

File: static/app/components/onboardingWizard/taskConfig.tsx (hunk 2 above)
Comment (EvanPurkhiser) [reply static/app/components/onboardingWizard/taskConfig.tsx:73]: yeah down for that

**Pull request merged**